### PR TITLE
vmware_guest_find: simplify the logic of the code

### PR DESCRIPTION
**Pull request by goneri**
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/297

##### SUMMARY

Simplify the logic of the `vmware_guest_find` module.

- reuse the existing `find_vm_by_name()` function.
- remove some unnecessary variables

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

vmware_guest_find

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_guest_find.py [hunk 1]
```diff
@@ -99,29 +99,17 @@ def __init__(self, module):
...
+        if self.uuid and self.use_instance_uuid:
+            vm_obj = find_vm_by_id(self.content, vm_id=self.uuid, vm_id_type="instance_uuid")
+        elif self.uuid:
+            vm_obj = find_vm_by_id(self.content, vm_id=self.uuid, vm_id_type="uuid")
         elif self.name:
-            objects = self.get_managed_objects_properties(vim_type=vim.VirtualMachine, properties=['name'])
-            for temp_vm_object in objects:
-                if temp_vm_object.obj.name == self.name:
-                    vms.append(temp_vm_object.obj)
+            vm_obj = find_vm_by_name(self.content, vm_name=self.name)
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest_find.py:110]: I would prefer `get_managed_objects_properties` over `find_vm_by_name`.

File: lib/ansible/modules/cloud/vmware/vmware_guest_find.py [hunk 2]
```diff
@@ -99,29 +99,17 @@ def __init__(self, module):
...
 
-        for vm in vms:
-            folder_path = self.get_vm_path(self.content, vm)
-            results.append(folder_path)
+        if vm_obj is None:
+            self.module.fail_json(msg="Failed to find the virtual machine with UUID: %s, name: %s" % (self.uuid, self.name))
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest_find.py:113]: `self.uuid` is None when name is specified.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_guest_find.py (hunk 2 above)
Comment (goneri) [reply lib/ansible/modules/cloud/vmware/vmware_guest_find.py:113]: I did that on purpose. I would prefer to only have one error message.

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by jctanner** [comment]
!waffling community_review

**Comment by jctanner** [comment]
!waffling needs_revision

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Pull request closed without merging**